src/toggle-switch/toggle-switch.scss

```diff
@@ -2,194 +2,212 @@
 // See LICENSE.md in the project root for license terms and full copyright notice.
 @import '../style/index';
 
+$toggle-switch-handle: $iui-icons-default;
+
 @mixin iui-toggle-switch {
   @include iui-reset;
   display: flex;
   align-items: center;
+  gap: $iui-s;
+  align-items: center;
   font-size: $iui-font-size;
   width: fit-content;
-  padding: ($iui-baseline/2) 0;
   user-select: none;
   cursor: pointer;
   position: relative;
+  isolation: isolate;
   @include themed {
     color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-2));
   }
 
-  > input {
-    width: 0;
-    height: 0;
+  // Disabled
+  &.iui-disabled {
+    cursor: not-allowed;
+    @include themed {
+      color: t(iui-text-color-muted);
+    }
+  }
+
+  // IE solution
+  // For labels on the right
+  > .iui-label {
+    margin-right: $iui-s;
+    @supports (gap: $iui-s) {
+      margin-right: 0;
+    }
+  }
+
+  // For labels on the left
+  .iui-toggle ~ .iui-label {
+    margin-left: $iui-s;
+    margin-right: 0;
+    @supports (gap: $iui-s) {
+      margin-left: 0;
+    }
+  }
+}
+
+@mixin iui-toggle {
+  display: grid;
```

Comment:
What is the point of using grid here if it is only one element?

Reply:
Using `grid` we can avoid using `position: absolute;` on `.iui-toggle-switch-icon`.

Reply:
For me it feels like a bigger hack than `position: absolute` 😀 Because you are saying in template that there will be one item - toggle -  but you are also using the same area name for icon.

Reply:
There is no need for named areas. In `input-with-icon` we're using the same idea with `grid-area: 1 / -1` for overlapping areas.

The benefit is that you get to leverage the centering from grid.

Reply:
> There is no need for named areas. In `input-with-icon` we're using the same idea with `grid-area: 1 / -1` for overlapping areas.

@FlyersPh9 Any actions for this?

Reply:
Yes, I collaborated with Mayank yesterday and we got this part working.